### PR TITLE
Add support for SAML authentication

### PR DESCRIPTION
**Pull request by zengyan-amazon**
*Issue #, if available:*

*Description of changes:* Add support for SAML authentication

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: server/backend/opendistro_security_client.ts [hunk 1]
```diff
@@ -136,9 +136,66 @@ export class SecurityClient {
...
     } catch (error) {
       throw new Error(error.message);
     }
   }
+
+  public async getSamlHeader(request: KibanaRequest) {
+    try {
+      const response = await this.esClient.asScoped(request).callAsCurrentUser('opendistro_security.authinfo');
+      // response is expected to be an error
+      throw Error('Invalid Saml configuration.');
```
Comment (yuxitang-amzn) [inline server/backend/opendistro_security_client.ts:149]: I didn't get the idea why we need to throw and catch the error in the same function. Could you provide some more information?

File: server/routes/index.ts [hunk 2]
```diff
@@ -50,7 +50,7 @@ export function defineRoutes(router: IRouter, esClient: IClusterClient) {
   });
 
   const actionGroupSchema = schema.object({
-    description: schema.string(),
+    description: schema.maybe(schema.string()),
```
Comment (yuxitang-amzn) [inline server/routes/index.ts:53]: This is not related to saml auth, right? Suggest move to a separate PR, and besides, please also remove `description` field for role/roleMapping/internalUser.

File: server/session/security_cookie.ts [hunk 3]
```diff
@@ -41,8 +47,23 @@ export function getSecurityCookieOptions(
...
+
+      // TODO: with setting redirect attributes to support OIDC and SAML,
+      //       we need to do additonal cookie validatin in AuthenticationHandlers.
+      // if SAML fields present
+      if (sessionStorage.saml && sessionStorage.saml.requestId && sessionStorage.saml.nextUrl) {
+        return { isValid: true, path: '/'};
+      }
+
+      // if OIDC fields present
+      if (sessionStorage.oidc ) {
```
Comment (yuxitang-amzn) [inline server/session/security_cookie.ts:62]: Out of curiosity, is OIDC related to SAML? If not, please add description for OIDC to commit message and CR description.

File: server/routes/index.ts (hunk 2 above)
Comment (zengyan-amazon) [reply server/routes/index.ts:53]: Yes, let me remove it and post and other PR for that

File: server/session/security_cookie.ts (hunk 3 above)
Comment (zengyan-amazon) [reply server/session/security_cookie.ts:62]: ah, no, OIDC has nothing to do with SAML, I can move this to another PR as well

File: server/backend/opendistro_security_client.ts (hunk 1 above)
Comment (zengyan-amazon) [reply server/backend/opendistro_security_client.ts:149]: this throw is no needed, the logic is that, call ES API, which expect an error, the error messages includes the direct location, and request ID of the SAML IDP
